core/store/src/trie/mod.rs

```diff
@@ -628,6 +629,13 @@ impl Trie {
         }
     }
 
+    pub fn retrieve_root_node(&self, root: &StateRoot) -> Result<(Vec<u8>, u64), StorageError> {
```

Comment:
After we merge `TrieNodeWithSize` and `StateRootNode` this method should return `Result<Option<TrieNodeWithSize>, StorageError>`. We should avoid encoding `Option` by passing `vec![]` and `0`.

Reply:
Even if `StateRootNode::data` does not always correspond to `TrieNodeWithSize` we still should return `Result<Option<TrieNodeWithSize>, StorageError>` here.

Reply:
I will return `StorageError::StorageInconsistentState` there.

Reply:
We should still avoid having `(Vec<u8>, u64)` around our code base for this specific scenario that happens all the time:

> When someone sees somewhere in the code `(Vec<u64>, u64)` they need to trace all the way through the code to the origin of this structure in order to understand what this type represents. This is time-consuming.

Reply:
++